src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Refactoring/RazorComponentRenameEndpoint.cs

```diff
@@ -0,0 +1,292 @@
+            foreach (var node in codeDocument.GetSyntaxTree().Root.DescendantNodes().Where(n => n.Kind == SyntaxKind.MarkupTagHelperElement))
+            {
+                if (node is MarkupTagHelperElementSyntax tagHelperElement && BindingsMatch(originTagHelperBinding, tagHelperElement.TagHelperInfo.BindingResult))
+                {
+                    documentChanges.Add(new WorkspaceEditDocumentChange(new TextDocumentEdit
+                    {
+                        TextDocument = documentIdentifier,
+                        Edits = CreateEditsForMarkupTagHelperElement(tagHelperElement, codeDocument, newName)
+                    }));
+                }
+            }
+        }
+
+        public List<TextEdit> CreateEditsForMarkupTagHelperElement(MarkupTagHelperElementSyntax element, RazorCodeDocument codeDocument, string newName)
+        {
+            var edits = new List<TextEdit>
+            {
+                new TextEdit()
+                {
+                    Range = element.StartTag.Name.GetRange(codeDocument.Source),
+                    NewText = newName,
+                },
+            };
+            if (element.EndTag != null)
+            {
+                edits.Add(new TextEdit()
+                {
+                    Range = element.EndTag.Name.GetRange(codeDocument.Source),
+                    NewText = newName,
+                });
+            }
+            return edits;
+        }
+
+        private static bool BindingsMatch(TagHelperBinding left, TagHelperBinding right)
+        {
+            foreach (var leftDescriptor in left.Descriptors)
+            {
+                foreach (var rightDescriptor in right.Descriptors)
+                {
+                    if (leftDescriptor.Equals(rightDescriptor))
+                    {
+                        return true;
+                    }
+                }
+            }
+            return false;
+        }
+
+        private async Task<TagHelperBinding> GetOriginTagHelperBindingAsync(DocumentSnapshot documentSnapshot, RazorCodeDocument codeDocument, Position position)
+        {
+            var sourceText = await documentSnapshot.GetTextAsync().ConfigureAwait(false);
+            var linePosition = new LinePosition((int)position.Line, (int)position.Character);
+            var hostDocumentIndex = sourceText.Lines.GetPosition(linePosition);
+            var location = new SourceLocation(hostDocumentIndex, (int)position.Line, (int)position.Character);
+
+            var change = new SourceChange(location.AbsoluteIndex, length: 0, newText: string.Empty);
+            var syntaxTree = codeDocument.GetSyntaxTree();
+            if (syntaxTree?.Root is null)
+            {
+                return null;
+            }
+
+            var owner = syntaxTree.Root.LocateOwner(change);
+            var node = owner.Ancestors().FirstOrDefault(n => n.Kind == SyntaxKind.MarkupTagHelperElement);
```

Comment:
Wont this let you attempt to rename a super-parent. Aka trying to rename "world" in "Hello World" below would result in it trying to rename `SomeComponent` I believe:

```
<SomeComponent>
    Hello World
</SomeComponent>
```

Reply:
Good call, thanks!